crypto/serializer/serializer_pkey.c

```diff
@@ -121,13 +121,13 @@ static void select_serializer(const char *name, void *data)
         return;
 
     if ((s = OSSL_SERIALIZER_fetch(d->libctx, name, d->propquery)) != NULL) {
-        if (d->first == NULL && s->serialize_data != NULL) {
-            d->first = s;
-        } else if (OSSL_SERIALIZER_provider(s) == d->desired_provider
+        if (OSSL_SERIALIZER_provider(s) == d->desired_provider
                    && s->serialize_object != NULL) {
```

Comment:
Nit: Even for you, this is a little too much...

Reply:
Fixed